diyepw/create_amy_epw_files_for_years_and_wmos.py

```diff
@@ -0,0 +1,83 @@
+import pandas as pd
+import os
+from typing import List, Tuple
+from .create_amy_epw_file import create_amy_epw_file
+from ._logging import _logger
+
+def create_amy_epw_files_for_years_and_wmos(
+        wmo_indices: List[int],
+        years: List[int],
+
+        *,
+        max_records_to_interpolate: int,
+        max_records_to_impute: int,
+        max_missing_amy_rows: int,
+        amy_epw_dir: str = None,
+        tmy_epw_dir: str = None,
+        amy_dir: str = None,
+        amy_files: Tuple[str, str] = None,
+        allow_downloads: bool = False
+) -> dict:
+    """
+    Create AMY EPW files for every combination of a set of WMO indices and years.
+
+    Except for "years" and "wmos" being lists rather than single values, all parameters are identical in effect to
+    create_amy_epw_file() - see that function's documentation for details.
+    :param wmo_indices:
+    :param years:
+    :param max_records_to_interpolate:
+    :param max_records_to_impute:
+    :param max_missing_amy_rows:
+    :param amy_epw_dir: Note that, in addition to the generated AMY EPW files, an errors.csv file will be created in this
+        directory if any errors were encountered, with error messages explaining which year/WMO Index combinations
+        failed and why.
+    :param tmy_epw_dir:
+    :param amy_dir:
+    :param amy_files:
+    :param allow_downloads:
+    :return: A dictionary of the files generated for each year/wmo combination, in the form {
+        <year>: {
+            <wmo_index>: [<file_path>, ...],
+            ...
+        },
+        ...
+    }
+    """
+    amy_epw_files = {}
+
+    # Initialize the df to hold information about year/WMO Index combinations for which no EPW could be generated
+    errors = pd.DataFrame(columns=['year', 'wmo_index', 'error'])
+    errors_path = os.path.join(amy_epw_dir, "errors.csv")
```

Comment:
this will raise an uncaught exception if `amy_epw_dir` is not provided, whereas `create_amy_epw_file` returns more. useful messages.

Reply:
having a default output folder that is gitignored could be nice

Reply:
Yeah, this was a bug. I've patched it by having it import the default location from the base create_amy_epw_files() method. I have it use a temporary directory instead of a local default folder because I don't want the user to have to go digging through their python installation looking for data files, something like /tmp/whatever.epw is nicer.